iodata/formats/json.py

```diff
@@ -0,0 +1,624 @@
+            del result["schema_name"]
+    if "schema_name" not in result:
+        # Attempt to determine schema type, since some QCElemental files omit this
+        warn(
+            "{}: QCSchema files should have a `schema_name` key."
+            "Attempting to determine schema type...".format(lit.filename),
+            FileFormatWarning,
+            2,
+        )
+        # Geometry is required in any molecule schema
+        if "geometry" in result:
+            schema_name = "qcschema_molecule"
+        # Check if BSE file, which is too different
+        elif "molssi_bse_schema" in result:
+            raise FileFormatError(
+                "{}: IOData does not currently support MolSSI BSE Basis JSON.".format(lit.filename)
+            )
+        # Center_data is required in any basis schema
+        elif "center_data" in result:
+            schema_name = "qcschema_basis"
+        elif "driver" in result:
+            if "return_result" in result:
+                schema_name = "qcschema_output"
+            else:
+                schema_name = "qcschema_input"
+        else:
+            raise FileFormatError("{}: Could not determine `schema_name`.".format(lit.filename))
+    else:
+        schema_name = result["schema_name"]
+    if "schema_version" not in result:
+        warn(
+            "{}: QCSchema files should have a `schema_version` key."
+            "Attempting to load without version number.".format(lit.filename),
+            FileFormatWarning,
+            2,
+        )
+
+    if schema_name == "qcschema_molecule":
+        return _load_qcschema_molecule(result, lit)
+    if schema_name == "qcschema_basis":
+        return _load_qcschema_basis(result, lit)
+    if schema_name == "qcschema_input":
+        return _load_qcschema_input(result, lit)
+    if schema_name == "qcschema_output":
+        return _load_qcschema_output(result, lit)
+    raise FileFormatError(
+        "{}: Invalid QCSchema type {}, should be one of `qcschema_molecule`, `qcschema_basis`,"
+        "`qcschema_input`, or `qcschema_output".format(lit.filename, result["schema_name"])
+    )
+
+
+def _load_qcschema_molecule(result: dict, lit: LineIterator) -> dict:
+    """Load qcschema_molecule properties.
+
+    Parameters
+    ----------
+    result
+        The JSON dict loaded from file, same as the `molecule` key in QCSchema input/output files.
+    lit
+        The line iterator holding the file data.
+
+    Returns
+    -------
+    molecule_dict
+        Output dictionary containing ``atcoords``, ``atnums``, ``charge``,
+        ``nelec`` & ``spinpol`` keys and corresponding values.
+        It may contain ``atmasses``, ``bonds``, ``g_rot``, ``title`` & ``extra``
+        keys and corresponding values as well.
+
+    """
+    # All Topology properties are found in the "molecule" key
+    molecule_dict = _parse_topology_keys(result, lit)
+
+    return molecule_dict
+
+
+def _parse_topology_keys(mol: dict, lit: LineIterator) -> dict:
```

Comment:
What do you mean by "topology"? Is this QCSchema terminology?

Reply:
Topology is essentially all the specifiers for a molecule. I'm using "topology", in line with how QCSchema does, because while the Molecule schema only uses these keys, the Input and Output schema also specify a molecule and will use this function to parse those keys.

Reply:
Thanks for explaining!